python/lsst/afw/geom/testUtils.py

```diff
@@ -1120,3 +1120,8 @@ def checkPersistence(self, transform):
 
         # Check pickling
         self.assertTransformsEqual(transform, pickle.loads(pickle.dumps(transform)))
+
+        # Check afw::table::io persistence round-trip
+        with lsst.utils.tests.getTempFilePath(".fits") as filename:
+            transform.writeFits(filename)
+            self.assertTransformsEqual(transform, type(transform).readFits(filename))
```

Comment:
Can you add function like the existing `transformFromString` that figures out the type and returns it?

Reply:
Will do.

Reply:
Actually, where is that existing `transformFromString` defined?  Is it possible it's not on master yet?

Reply:
It is a pure python function defined in lsst/afw/geom/transformFromString.py (I didn't see much point to trying to make a C++ version). It has been on master for a long time. I'm not sure how easily it can be emulated for this case because the reading is done in C++ and one has to read some of the data to figure out the data type to return. That doesn't seem very practical in C++.

Reply:
Yeah, I think it'd be tricky to implement.  Here are my thoughts on how best to do it:
 - If you call `table.io.Persistable.readFits` on a FITS file that contains a `Transform`, it will already return an opaque object that actually holds a `Transform` of the right type.  It'd be easy to call that within a function with a more intuitive name.
 - *If* we could expose to pybind11 the fact that `Transform` inherits from `table.io.Persistable`, pybind11 would automatically cast that opaque object to the right kind of `Transform` on return - but we can't expose that inheritance to pybind11 because hiding it is how we avoid circular dependency problems.
 - We could get the same behavior by adding a non-templated polymorphic base class between `Persistable` and the `Transform` templates (even if it doesn't have any methods); that could be exposed to pybind11, and if we then wrap a trivial C++ function that returns a (say) `TransformBase` from a FITS file, pybind11 will automatically downcast to the right derived/templated type.

I don't think that's worthwhile just for this function, but if `TransformBase` would be useful for other purposes, it might be worth considering.  Because `Transform`'s most important methods depend on its template parameters, `TransformBase` can't expose a very useful interface, but it could have a few pure virtual methods: `hasForward`, `hasInverse`, and `writeStream`.  It would simplify the implementation of the current `transformFromString` (since that could just wrap a C++ function that returns `TransformBase`, and pybind11 would do the casting there, too).  Let me know what you think.

Reply:
Interesting idea. I agree it's probably not worth the bother, but feel free to do it if you are interested. I am not sure how much use we would get out of the capability (we have no users of `transformFromString` yet, as far as I can tell).

Reply:
Ok, I'll skip it for now.  Should probably focus on things we know we need rather than things we think we might need.